packages/core/src/App/Containers/Layout/header/dashboard-header.jsx

```diff
@@ -55,60 +50,94 @@ const LoggedInHeader = ({ is_dark_mode }) => {
 const LoggedOutHeader = () => {
     const history = useHistory();
     const { is_dashboard } = React.useContext(PlatformContext);
+    const [active_dropdown, setCurrentDropdown] = React.useState('');
+    const [active_link_ref, setActiveLinkRef] = React.useState(null);
+    const [active_dropdown_ref, setActiveDropdownRef] = React.useState(null);
+    const nav_dropdown_ref = React.useRef(null);
+
+    const handleDropdownClick = (dropdown, target) => {
+        setCurrentDropdown(dropdown);
+        if (target) {
+            setActiveLinkRef(target);
+        }
+    };
+
+    const handleOutsideClick = e => {
```

Comment:
Same here, since this is created on each render your cleanup effect (lines 76-78) will probably not work. To fix this you can wrap it in a `React.useCallback` to guarantee the reference to this function is correct.

Reply:
Noted, amended both